### PR TITLE
Couple Minor Fixes

### PR DESCRIPTION
**Pull request by MicroDev1**

### CODE REVIEW

File: ports/espressif/.gitignore [hunk 1]
```diff
@@ -1,2 +1,2 @@
-build*/
-sdkconfig.old
+# lock files for examples and components
+dependencies.lock
```
Comment (dhalbert) [inline ports/espressif/.gitignore:2]: Did you mean to remove `build*/` and `sdkconfig.old`?

**Review by dhalbert — CHANGES_REQUESTED** [review]

File: ports/espressif/.gitignore (hunk 1 above)
Comment (MicroDev1) [reply ports/espressif/.gitignore:2]: I haven't noticed those appearing with build commands mentioned in `Makefile`.

Comment (dhalbert) [reply ports/espressif/.gitignore:2]: I have `build-*` directories:
```sh
~/repos/circuitpython/ports/espressif$ ls -d build*
build-adafruit_feather_esp32s2  build-adafruit_feather_esp32s2_tft  build-adafruit_feather_esp32_v2  build-adafruit_qtpy_esp32_pico
```
`sdkconfig.old` files get created by running `idf.py menuconfig`

Comment (MicroDev1) [reply ports/espressif/.gitignore:2]: Following are excluded in the CP root `.gitignore`.
```
build/
build-*/
```

Comment (MicroDev1) [reply ports/espressif/.gitignore:2]: I'll readd `sdkconfig`.

Comment (jepler) [reply ports/espressif/.gitignore:5]: what creates dependencies.lock?

Comment (MicroDev1) [reply ports/espressif/.gitignore:5]: > what creates dependencies.lock?

It's the [IDF Component Manager](https://docs.espressif.com/projects/esp-idf/en/latest/api-guides/tools/idf-component-manager.html).

Comment (dhalbert) [reply ports/espressif/.gitignore:5]: Thanks, didn't know `build*` was in the root `.gitignore`. It's the only entry in a bunch of `ports/*/.gitignore` files.

Comment (dhalbert) [reply ports/espressif/.gitignore:5]: I will add some commits to this PR to remove the redundant `port/` .gitignore's, unless you are already on it.

Comment (MicroDev1) [reply ports/espressif/.gitignore:5]: > I will add some commits to this PR to remove the redundant `port/` .gitignore's...

Sure👍

**Comment by dhalbert** [comment]
I pushed a new commit that removed a bunch of redundant `port/*/.gitignore` files, and removed some stale entries from others. Also `ports/unix/.gitignore` was ignoring `*.py` files, but now there are some in `variants/`. I did a `unix` build to check that no build-time `*.py` were created

**Review by dhalbert — APPROVED** [review]
Thanks for noticing these!

**Pull request merged**